server/src/main/java/io/crate/planner/optimizer/rule/RewriteFilterOnOuterJoinToInnerJoin.java

```diff
@@ -273,11 +275,23 @@ private static boolean couldMatchOnNull(@Nullable Symbol query,
         if (query == null) {
             return false;
         }
-        Symbol queryWithNulls = RefReplacer.replaceRefs(
-            FieldReplacer.replaceFields(query, ignored -> Literal.NULL),
-            ignored -> Literal.NULL
-        );
+        var visitor = new FunctionCopyVisitor<Void>() {
```

Comment:
Please use the introduced `NullSymbolEvaluator`, you just need to add the `visitOuterColumn()` I think.